pkg/provisioning/service.go

```diff
@@ -102,6 +102,20 @@ func (s *Service) Init(ctx context.Context) error {
 		allPls = append(allPls, duplicateID)
 	}
 
+	// remove pipelines with duplicate IDs from API pipelines
+	var existingPipelinesFromAPI []int
+	for i, pl := range configs {
+		_, err := s.pipelineService.Get(ctx, pl.ID)
+		// if err is nil then a pipeline exists already, so we add towards duplicate ID list
+		if err == nil {
+			multierr = cerrors.Errorf("pipelines with ID %q will be skipped: %w", pl.ID, ErrDuplicatedAPIPipelineID)
```

Comment:
We should append errors together: `multierror.Append(multierr, err)`. Same thing on L114 below.